src/game/room.c

```diff
@@ -661,13 +661,14 @@ void Room_AlterFloorHeight(ITEM_INFO *item, int32_t height)
         sector = &r->sectors[z_sector + x_sector * r->z_size];
     }
 
-    if (sector->floor != NO_HEIGHT / 256) {
-        sector->floor += height >> 8;
-        if (sector->floor == sky_sector->ceiling) {
-            sector->floor = NO_HEIGHT / 256;
+    if (sector->floor.height != NO_HEIGHT) {
+        sector->floor.height += height & ~(STEP_L - 1);
+        if (sector->floor.height == sky_sector->ceiling.height) {
+            sector->floor.height = NO_HEIGHT;
         }
     } else {
-        sector->floor = sky_sector->ceiling + (height >> 8);
+        sector->floor.height =
+            sky_sector->ceiling.height + (height & ~(STEP_L - 1));
```

Comment:
is there any way we can make this less ugly?

Reply:
We could introduce a macro for this, such as round to click or something.